selfservice/strategy/code/strategy_recovery.go

```diff
@@ -545,7 +545,12 @@ func (s *Strategy) recoveryHandleFormSubmission(w http.ResponseWriter, r *http.R
 	f.Active = sqlxx.NullString(s.RecoveryNodeGroup())
 	f.State = recovery.StateEmailSent
 	f.UI.Messages.Set(text.NewRecoveryEmailWithCodeSent())
-	f.UI.Nodes.Append(node.NewInputField("code", nil, node.CodeGroup, node.InputAttributeTypeText, node.WithRequiredInputAttribute).
+	f.UI.Nodes.Append(node.NewInputField("code", nil, node.CodeGroup, node.InputAttributeTypeText, node.WithInputAttributes(func(a *node.InputAttributes) {
+		a.Required = true
+		// Unique key for React to make sure input field is refreshed
+		a.Key = "recoveryKey"
```

Comment:
This would be the first instance where framework requirements (e.g. the React key) end up in business logic in Ory Kratos. I'd appreciate it if we could remove this. Pattern and required are fine :)

Reply:
Yes, of course. I removed those React-related items and now only Pattern added. Have to think of another way to fix the UI issue.